controllers/rcj_soccer_player_b1/rcj_soccer_player_b1.py

```diff
@@ -0,0 +1,62 @@
+# rcj_soccer_player controller - ROBOT B1
+
+###### REQUIRED in order to import abstract robot class
+import sys
+from pathlib import Path
+sys.path.append(str(Path('.').absolute().parent))
+from abstract_robot import AbstractRobot, TIME_STEP
+######
+
+import math
+
+
+class MyRobot(AbstractRobot):
+    def run(self):
+        while self.robot.step(TIME_STEP) != -1:
+            if self.is_new_data():
+                data = self.get_new_data()
+
+                # Get the position of our robot
+                robot_pos = data[self.name]
+                # Get the position of the ball
+                ball_pos = data['ball']
+
+                robot_angle = robot_pos['orientation']
+
+                # Get the angle between the robot and the ball
+                angle = math.atan2(ball_pos['y'] - robot_pos['y'],
+                                   ball_pos['x'] - robot_pos['x'])
+
+                if angle < 0:
+                    angle = 2 * math.pi + angle
+
+                if robot_angle < 0:
+                    robot_angle = 2 * math.pi + robot_angle
+
+                degrees = math.degrees(angle + robot_angle)
+
+                # Axis Z is forward
+                # TODO: change the robot's orientation so that X axis means forward
+                degrees -= 90
+                if degrees > 360:
+                    degrees -= 360
```

Comment:
@mrshu I wanted to keep the abstract class clean, so there is no game logic involved -- and use controllers as a template. I understand your point that the code is copied throughout the robot controllers, but it might confuse participants about how to implement it.
What do you think about this?

Reply:
Sorry @Adman, I was not clear enough in here.

What I meant is that this whole business about figuring out the angle between the robot and the ball and then their normalization could be abstracted out. I don't see it as part of the game logic -- I would expect the controller's sample code to just call a function that would compute these. Something like

```python
# Get the position of our robot
robot_pos = data[self.name]
# Get the position of the ball
ball_pos = data['ball']

ball_angle_deg, robot_angle_deg = self.get_angles(ball_pos, robot_pos)
```

Would that make sense?

Reply:
Yes, makes sense. Going implement that.